### PR TITLE
feat: added preferences page

### PR DESCRIPTION
**Pull request by CzarekDryl**
Added preferences page

https://tw.auroracreation.com/app/tasks/15529399

image

### CODE REVIEW

File: src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx [hunk 1]
```diff
@@ -1,23 +1,182 @@
...
+                  <div className="flex items-center justify-end">
+                    <span className="text-md mr-6">{emailValue}</span>
+                    <Button
+                      mode="primarySolid"
+                      text={{ id: 'button.updateEmail' }}
+                      onClick={() => setIsEmailInputVisible(true)}
+                    />
+                  </div>
+                )}
+                {!emailValue || isEmailInputVisible ? (
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx:60]: {!emailValue || isEmailInputVisible && (...)}

File: src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import { useAppContext, useCanEditProfile } from '~hooks';
+import Toast from '~shared/Extensions/Toast';
+
+export const useUserPreferencesPage = () => {
+  const { formatMessage } = useIntl();
+  const { updateUser } = useAppContext();
+  const [editUser] = useUpdateUserProfileMutation();
+  const { user } = useCanEditProfile();
+
+  const EMAIL_REGEX = /^[\w-.]+@([\w-]+\.)+[\w-]+$/;
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx:20]: please export and use regex from here: src/components/common/CreateUserWizard/validation.ts

File: src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  const isValidEmail = (email: string) => {
+    return email ? new RegExp(EMAIL_REGEX).test(email) : false;
+  };
+  const isEmailAlreadyRegistered = createYupTestFromQuery({
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx:24]: please use method from here: src/components/common/CreateUserWizard/validation.ts

File: src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      await editUser({
+        variables: {
+          input: {
+            id: user?.walletAddress || '',
+            ...updatedProfile,
+            email: updatedProfile.email || null,
+          },
+        },
+      });
+
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx:71]: is there no toast on success needed?

File: src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx [hunk 5]
```diff
@@ -1,23 +1,182 @@
...
+              </div>
+            </div>
+            <span className="divider" />
+            <div className={styles.row}>
+              <div className="flex md:flex-row flex-col items-center justify-between p-3 bg-gray-50 rounded-lg w-full">
+                <div className="flex items-center mb-3 md:mb-0">
+                  <Icon name="cardholder" appearance={{ size: 'small' }} />
+                  <span className="text-md ml-2 truncate block w-full">
+                    {isMobile
+                      ? multiLineTextEllipsis(user.walletAddress, 20)
```
Comment (krawczykkatarzyna) [inline src/components/frame/v5/pages/UserPreferencesPage/UserPreferencesPage.tsx:87]: maybe better will be to pass the second parameter as a optional props added to UserPreferencesPage component and with default value 20?

File: src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx (hunk 4 above)
Comment (CzarekDryl) [reply src/components/frame/v5/pages/UserPreferencesPage/hooks.tsx:62]: Toast added

File: src/i18n/en.json [hunk 6]
```diff
@@ -748,18 +754,38 @@
...
+  "field.display": "Enable action and motion windows as full screen",
+  "description.display": "As default, all action and motion windows appear as a side panel",
+  "field.switchAccount": "Switch account",
+  "description.switchAccount": "Connect with a different wallet",
+  "field.emailNotification": "Enable email notifications",
+  "description.emailNotification": "If enabled, we’ll send you information about Colony’s new product features and updates",
+  "field.email": "Email address",
+  "description.email": "Update your email address",
   "description.location": "Your location is optional and only shown on your Colony profile",
+  "button.connect.discord": "Connect Discord",
```
Comment (KatarzynaStefanowiczPagepro) [inline src/i18n/en.json:774]: `button.connectDiscord` and move to the button section

**Review by KatarzynaStefanowiczPagepro — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Pull request merged**